R/pkg/R/types.R

```diff
@@ -83,3 +83,38 @@ specialtypeshandle <- function(type) {
   }
   returntype
 }
+
+# Helper function that checks supported types in Arrow.
+checkSchemaInArrow <- function(schema) {
+  stopifnot(inherits(schema, "structType"))
```

Comment:
nit: this is kinda duplicated in the caller - yes but either we trust they are calling it right or we double check here.